File: Back/Auction/src/main/java/tn/esprit/auction/Entites/Room.java
package tn.esprit.auction.Entites;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="idRoom")
    private Long idRoom;
    private String codeRoom;
    private float price;
    private TypePack typePack ;
    private Date dateDebut;
    private int duration ;
    private int maxParticipants ;
    private String description ;
    private Boolean status ;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "room")
    private List<Pack> packages;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "room")
    private List<Company> companies;

}

File: Front/projects/back-office/src/app/Models/Room.ts
import { Pack } from "./Pack";

export class Room{

    idRoom!:number;
    codeRoom!:string;
    quantity!:number;
    price!:number;
    priceAuction!:number;
    typePack!:string;
    dateDebut!:Date;
    duration !:number;
    description!:string
    status !:boolean;
    maxParticipants!:number;
    confirmedParticipant!:number;
    packages!:Pack[]
}